### PR TITLE
Multi-window support

### PR DESCRIPTION
**Pull request by andyduke**
Part of the Android player is rewritten to support the multi-window mode. Now, when you turn on the multi-window mode and changing the size of the player window, the video and sound continues without breaks.

The example application is updated - support for the multi-window is added to the AndroidManifest.

### CODE REVIEW

File: flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayer.java [hunk 1]
```diff
@@ -647,4 +659,231 @@ private void log(String message) {
         }
     }
 
+    // --- VLCTextureView
+
+    public class VLCTextureView extends TextureView implements TextureView.SurfaceTextureListener, View.OnLayoutChangeListener, IVLCVout.OnNewVideoLayoutListener {
```
Comment (solid-software) [inline flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayer.java:664]: Do you mind moving it to a separate file?

**Review by solid-software — CHANGES_REQUESTED** [review]
Thanks for the PR. I've added a styling suggestion please let me know if you can fix it.

**Comment by solid-software** [comment]
@alr2413 @mitchross do you guys want to check this PR?

File: flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayer.java (hunk 1 above)
Comment (andyduke) [reply flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/FlutterVlcPlayer.java:664]: Yes, it's reasonable.

File: flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    }
+
+    public VLCTextureView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        mContext = context;
+        initVideoView();
+    }
+
+    public void dispose() {
+        log("************************** dispose");
```
Comment (mitchross) [inline flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java:51]: remove this or add a log debug

File: flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                setSurfaceTexture(texture);
+            }
+        }
+    }
+
+    @Override
+    public void onSurfaceTextureAvailable(SurfaceTexture surface, int width, int height) {
+        log("************************** onSurfaceTextureAvailable");
+
+        if (mSurfaceTexture == null || mSurfaceTexture.isReleased()) {
```
Comment (mitchross) [inline flutter_vlc_player/android/src/main/java/software/solid/fluttervlcplayer/VLCTextureView.java:116]: Its better to close the second } if you are going to start with { 

its alot easier to read open and close brackets.

**Comment by mitchross** [comment]
@solid-software  this is fine to merge .

**Comment by solid-software** [comment]
Thanks @mitchross! @andyduke do you mind adding changelog for this PR? Basically updating this file https://github.com/solid-software/flutter_vlc_player/blob/master/flutter_vlc_player/CHANGELOG.md and pubspec?

**Comment by andyduke** [comment]
> Thanks @mitchross! @andyduke do you mind adding changelog for this PR? Basically updating this file https://github.com/solid-software/flutter_vlc_player/blob/master/flutter_vlc_player/CHANGELOG.md and pubspec?

I changed the version in the pubspec, added the version description to the CHANGELOG, and added to the README file a description of how to configure the Android app to enable multi-window support.

**Review by solid-software — APPROVED** [review]
LGTM - thanks @andyduke !

**Pull request merged**